synapse/config/emailconfig.py

```diff
@@ -509,6 +474,63 @@ def generate_config_section(self, config_dir_path, server_name, **kwargs):
             % DEFAULT_SUBJECTS
         )
 
+    def read_templates(
+        self,
+        filenames: List[str],
+        custom_template_directory: Optional[str] = None,
+        autoescape: bool = True,
+        filters: Dict[str, Callable] = {},
+    ) -> List[jinja2.Template]:
+        """Load a list of template files from disk using the given variables and filters.
+
+        This function will attempt to load the given templates from the default Synapse
+        template directory. If `custom_template_directory` is supplied, that directory
+        is tried first.
+
+        Files read are treated as Jinja templates. These templates are not rendered yet.
+
+        Args:
+            filenames: A list of template filenames to read.
+
+            custom_template_directory: A directory to try to look for the templates
+                before using the default Synapse template directory instead.
+
+            autoescape: Whether to automatically escape template variables when the templates
+                are rendered.
+
+            filters: Custom functions which can modify the given template variables upon
+                render, if the template file specifies them.
+
+        Raises:
+            ConfigError: if the file's path is incorrect or otherwise cannot be read.
+
+        Returns:
+            A list of jinja2 templates.
+        """
+        templates = []
+        for filename in filenames:
+            template_directory = self.default_template_dir
+
+            # Check whether the template exists in the user-configured template directory
+            if custom_template_directory:
+                filepath = os.path.join(template_directory, filename)
+                if self.path_exists(filepath):
+                    # Use the custom template directory instead
+                    template_directory = custom_template_directory
+
+            # Set up jinja with the template directory
+            loader = jinja2.FileSystemLoader(template_directory)
```

Comment:
It looks like [`FileSystemLoader`](https://jinja.palletsprojects.com/en/2.11.x/api/#jinja2.FileSystemLoader) can take multiple locations which it will try in order, this would allow us to have a single file system loader that gets re-used (and avoid having to check each time we try to load a file).

Reply:
Ah, TIL!